lib/improver/tests/metadata/test_set_attributes.py

```diff
@@ -0,0 +1,143 @@
+
+    def setUp(self):
+        """Set up a dummy cube"""
+        attributes = {
+            "source": "Met Office Unified Model",
+            "title": "MOGREPS-UK Temperature Forecast "
+                     "on UK 2 km Standard Grid"}
+        self.cube = set_up_variable_cube(
+            278*np.ones((5, 5), dtype=np.float32), attributes=attributes)
+
+        self.blend_descriptor = "IMPROVER Post-Processed Multi-Model Blend"
+        self.nowcast_descriptor = "MONOW Extrapolation Nowcast"
+        self.grid_descriptor = " on UK 2 km Standard Grid"
+
+    def test_blend(self):
+        """Test blended attributes"""
+        expected_title = self.blend_descriptor + self.grid_descriptor
+        result = set_product_attributes(self.cube, "multi-model blend")
+        attrs = result.attributes
+        self.assertEqual(attrs["source"], "IMPROVER")
+        self.assertEqual(attrs["title"], expected_title)
+        self.assertEqual(attrs["institution"], "Met Office")
+
+    def test_nowcast(self):
+        """Test nowcast attributes"""
+        expected_title = self.nowcast_descriptor + self.grid_descriptor
+        result = set_product_attributes(self.cube, "nowcast")
+        attrs = result.attributes
+        self.assertEqual(attrs["source"], "IMPROVER")
+        self.assertEqual(attrs["title"], expected_title)
+        self.assertEqual(attrs["institution"], "Met Office")
+
+    def test_no_grid_descriptor(self):
+        """Test blended cube only has grid descriptor where appropriate"""
+        self.cube.attributes["title"] = "MOGREPS-UK Temperature Forecast"
+        result = set_product_attributes(self.cube, "multi-model blend")
+        self.assertEqual(result.attributes["title"], self.blend_descriptor)
+
+    def test_error_invalid_product(self):
+        """Test error when product is not recognised"""
+        msg = "product 'unknown product' not available"
+        with self.assertRaisesRegex(ValueError, msg):
+            set_product_attributes(self.cube, "unknown product")
+
+
+class Test_update_spot_title_attribute(unittest.TestCase):
+    """Test the update_spot_title_attribute function"""
+
+    def setUp(self):
+        """Set up a test cube (spot data not required)"""
+        attributes = {
+            "title": "MOGREPS-UK Temperature Forecast "
+                     "on UK 2 km Standard Grid"}
+        self.cube = set_up_variable_cube(
+            278*np.ones((2, 2), dtype=np.float32), attributes=attributes)
+        self.expected_title = "MOGREPS-UK Temperature Forecast UK Spot Values"
+
+    def test_basic(self):
+        """Test title is correctly updated in place"""
+        update_spot_title_attribute(self.cube)
+        self.assertEqual(self.cube.attributes["title"], self.expected_title)
+
+    def test_other_grid(self):
+        """Test function recognises other grid specifications"""
+        self.cube.attributes["title"] = (
+            "MOGREPS-UK Temperature Forecast on Other Grid")
```

Comment:
I'm a bit surprised that "MOGREPS-UK Temperature Forecast on Other Grid" as input would return "MOGREPS-UK Temperature Forecast UK Spot Values", based on the fact that there's a "UK" in "MOGREPS-UK". Why is this the intended behaviour?

Reply:
Ah it's not the intended behaviour.  I wanted the code to look for "UK" in the "grid" part of the regular expression.  I'll look at this again and figure out how to make that work.  Thanks for the spot.

Reply:
Fixed.